pkg/autodiscovery/listeners/snmp.go

```diff
@@ -228,7 +228,7 @@ func (l *SNMPListener) checkDevices() {
 	discoveryTicker := time.NewTicker(time.Duration(l.config.DiscoveryInterval) * time.Second)
 
 	for {
-		for _, subnet := range subnets {
+		for i, subnet := range subnets {
```

Comment:
good catch,
to avoid `subnet` copy what you can also do

```suggestion
		var subnet *snmpSubnet
		for i := range subnets {
		    subnet = &subnets[i]
```

Reply:
Good point. Fixed here https://github.com/DataDog/datadog-agent/pull/6943/commits/8267b1fbf412394924fd1675995e945901b5a2a3

Reply:
@clamoriniere Do you mind 👀  again?